core/src/main/java/org/opensearch/sql/calcite/CalciteRelNodeVisitor.java

```diff
@@ -620,6 +623,20 @@ public RelNode visitDedupe(Dedupe node, CalcitePlanContext context) {
     return context.relBuilder.peek();
   }
 
+  @Override
+  public RelNode visitWindow(Window node, CalcitePlanContext context) {
+    visitChildren(node, context);
+
+    for (UnresolvedExpression window : node.getWindowFunctionList()) {
+      ((WindowFunction) ((Alias) window).getDelegated()).setPartitionByList(node.getPartExprList());
```

Comment:
Could you elaborate the case that Window has a list of window function? And in this case, we keep overriding the partitionBy list by setter?

Reply:
`Window` plan actually similar to `Eval` which is adding expressions to project. For example,
```
SELECT *, count(*) over(partition by state) as cnt, avg(age) over(partition by state) as avg, min(age) over(partition by state) as min, max(age) over(partition by state) as max
FROM state_country;
```
create 4 window functions:
```
Window (count over, avg over, min over, max over)
    |- Project(*)
        |-Relation(state_country)
```
It will be resolve to
```
Project(*, count over, avg over, min over, max over)
  |-Relation(state_country)
```